lib/cli/tests.rs

```diff
@@ -399,11 +399,13 @@ fn should_fail_to_create_transfer_with_no_secret_key_while_not_allowing_unsigned
     );
 
     assert!(transfer_deploy.is_err());
+    let _error_string =
+        "allow_unsigned_deploy was false, but no secret key was provided".to_string();
     assert!(matches!(
         transfer_deploy.unwrap_err(),
         CliError::InvalidArgument {
-            context: "new_transfer (secret_key, allow_unsigned_deploy)",
-            error: _
+            context: "new_transfer",
+            error: _error_string,
         }
```

Comment:
this is not matching on the actual value of the `_error_string`, it's just binding a new `_error_string` variable, this should either match on the actual string literal or add an `if` guard to the match:
`matches!(..., CliError::InvalidArgument {..., error} if error == expected`

Reply:
Thanks good catch I don't remember why I changed that probably linting. If this creates a binding instead of what I meant indeed it's always true. I think my eyes probably read assert_eq! which I use more often which would have kind of worked I guess with Partial Eq. I see I can't fool you with a badly used matches.